### PR TITLE
Add phi3 support in ONNX exporter

### LINKED ISSUE #1826: Phi3 support
### Feature request

Microsoft's new phi3 mode, in particular the 128K context mini model, is not supported by Optimum export. 

Error is:
"ValueError: Trying to export a phi3 model, that is a custom or unsupported architecture, but no custom export configuration was passed as `custom_export_configs`. Please refer to https://huggingface.co/docs/optimum/main/en/exporters/onnx/usage_guides/export_a_model#custom-export-of-transformers-models for an example on how to export custom models. Please open an issue at https://github.com/huggingface/optimum/issues if you would like the model type phi3 to be supported natively in the ONNX export."

### Motivation

Phi3-mini is potentially very significant as it has a large context but a small size. This could be used in lots of scenarios if it has good performance.

### Your contribution

Unlikely I could do a PR as ONNX work is not my forte.

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

as per title

Fixes #1826 

## Example

(Needs transformers>=4.41.0)

* CLI

```bash
optimum-cli export onnx --model microsoft/Phi-3-mini-4k-instruct phi3_onnx/
```

* ORTModel

```python
from transformers import AutoTokenizer

from optimum.onnxruntime import ORTModelForCausalLM


# [Export]
model_id = "microsoft/Phi-3-mini-4k-instruct"
model = ORTModelForCausalLM.from_pretrained(model_id, export=True)
model.save_pretrained("phi3_onnx/")

# [Inference]
model_id = "phi3_onnx/"
tokenizer = AutoTokenizer.from_pretrained(model_id)
model = ORTModelForCausalLM.from_pretrained(model_id)
inputs = tokenizer(["Hello, my dog is cute"], return_tensors="pt")
outputs = model.generate(**inputs, max_length=50)
print(tokenizer.batch_decode(outputs, skip_special_tokens=True))
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/exporters/exporters_utils.py [hunk 1]
```diff
@@ -132,6 +132,7 @@
         "hf-internal-testing/tiny-random-vision_perceiver_conv": ["image-classification"],
     },
     "phi": "echarlaix/tiny-random-PhiForCausalLM",
+    "phi3": "katuni4ka/tiny-random-phi3",
```
Comment (echarlaix) [inline tests/exporters/exporters_utils.py:135]: what do you think about also adding a test to [test_modeling.py](https://github.com/huggingface/optimum/blob/e0f58121140ce4baa01919ad70a6c13e936f7605/tests/onnxruntime/test_modeling.py#L2248) using this model

**Review by echarlaix — APPROVED** [review]
LGTM, thanks @JingyaHuang !

File: tests/exporters/exporters_utils.py (hunk 1 above)
Comment (JingyaHuang) [reply tests/exporters/exporters_utils.py:135]: Added!

**Review by IlyasMoutawwakil — APPROVED** [review]
LGTM !

**Comment by JingyaHuang** [comment]
Merging since failed CIs are irrelevant to the phi3 support.

**Pull request merged**